server/controllers/v1/post.ts

```diff
@@ -340,35 +340,22 @@ export default class PostController {
     }
 
     if (tags) {
-      // could not get adding (multiple) tags to work any other way
-      const tagObjs = await Promise.all(
-        tags.slice(0, 3).map(async (text) => {
-          return this.setupTag(text.trim());
-        })
+      const tagObjs = await this.tagsRepo.bulkCreate(
+        // create (or find) our tag objects
+        tags.slice(0, 3).map((t) => {
+          // restrict to max 3 tags
+          return { text: t.trim() };
+        }),
+        {
+          ignoreDuplicates: true,
+        }
       );
-      await post.addTags(tagObjs);
+      await post.addTags(tagObjs); // junction table association
```

Comment:
```suggestion
      await post.addTags(tagObjs);
```
Comments should usually explain why you did something the way you did it, not what's happening.